src/libstore/local-fs-store.cc

```diff
@@ -99,6 +99,13 @@ std::optional<std::string> LocalFSStore::getBuildLog(const StorePath & path_)
         } catch (InvalidPath &) {
             return std::nullopt;
         }
+    } else if (settings.isExperimentalFeatureEnabled(Xp::CaDerivations)) {
+        // The build log is actually attached to the corresponding resolved
+        // derivation, so we need to get it first
+        auto drv = readDerivation(path);
+        auto resolvedDrv = drv.tryResolve(*this);
+        if (resolvedDrv)
+            path = writeDerivation(*this, *resolvedDrv);
```

Comment:
In fact the `!path.isDerivation()` part is also duplicated between the two stores that implement this, too. That could also go in `getBuildLogResolved` (or whatever the wrapper function should be called).

Reply:
I actually did this intentionnally, because as-it-is, the logic assumes that the derivation is there (we can’t resolve without reading it), so that would break with most binary caches

Reply:
I see, but can't we just see if the derivation exists and only try resolving it if it does?